neqo-transport/src/connection.rs

```diff
@@ -1751,6 +1784,18 @@ mod tests {
         assert!(out.as_dgram_ref().is_some());
         qdebug!("Output={:0x?}", out.as_dgram_ref());
 
+        qdebug!("---- client: cert verification");
+        let out = client.process(out.dgram(), now());
+        assert!(out.as_dgram_ref().is_some());
+        qdebug!("Output={:0x?}", out.as_dgram_ref());
+
+        let out = server.process(out.dgram(), now());
+        assert!(out.as_dgram_ref().is_none());
+
+        let authentication_needed = |e| matches!(e, ConnectionEvent::AuthenticationNeeded);
+        assert!(client.events().into_iter().any(authentication_needed));
```

Comment:
Can you run `cargo clippy --tests` and fix anything that comes up?  Assuming that #122 is merged ahead of this, that is.  Otherwise, I think that you need to do this (and elsewhere as well):

```suggestion
        assert!(client.events().any(authentication_needed));
```